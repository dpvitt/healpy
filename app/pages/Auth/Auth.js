import React, { useState } from 'react';
import { Auth as AWSAuth } from 'aws-amplify';
import { View } from 'react-native';

import { SignIn } from './SignIn';
import { SignUp } from './SignUp';
import { VerifyModal } from './VerifyModal';

export const Auth = ({ navigation: { navigate } }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [confirmationCode, setConfirmationCode] = useState('');
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <SignUp
        setEmail={setEmail}
        setPassword={setPassword}
        setConfirmPassword={setConfirmPassword}
        handleSignUp={() =>
          handleSignUp({ email, password, confirmPassword, setModalVisible })
        }
      />
      <SignIn
        setEmail={setEmail}
        setPassword={setPassword}
        handleSignIn={() => handleSignIn({ email, password, navigate })}
      />
      <VerifyModal
        modalVisible={modalVisible}
        setConfirmationCode={setConfirmationCode}
        handleConfirmationCode={() =>
          handleConfirmationCode({
            email,
            confirmationCode,
            setModalVisible,
            navigate,
          })
        }
      />
    </View>
  );
};

const handleSignIn = ({ email, password, navigate }) => {
  AWSAuth.signIn(email, password)
    .then((user) => {
      console.log(user);
      navigate('Home');
    })
    .catch((err) => console.log(err));
};

const handleSignUp = ({
  email,
  password,
  confirmPassword,
  setModalVisible,
}) => {
  if (password === confirmPassword) {
    AWSAuth.signUp({
      username: email,
      password,
      attributes: { email },
    })
      .then(() => setModalVisible(true))
      .catch((err) => console.log(err));
  }
};

const handleConfirmationCode = ({
  email,
  confirmationCode,
  setModalVisible,
  navigate,
}) => {
  AWSAuth.confirmSignUp(email, confirmationCode, {})
    .then(() => {
      setModalVisible(false);
      navigate('Home');
    })
    .catch((err) => console.log(err));
};
