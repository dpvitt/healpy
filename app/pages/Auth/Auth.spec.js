import React from 'react';
import { shallow } from 'enzyme';
import { Auth } from './Auth';
import { Auth as AWSAuth } from 'aws-amplify';

jest.mock('aws-amplify', () => ({
  Auth: {
    signIn: jest.fn().mockResolvedValue(),
    signUp: jest.fn().mockResolvedValue(),
    confirmSignUp: jest.fn().mockResolvedValue(),
  },
}));

describe('<Auth />', () => {
  const props = { navigation: { navigate: jest.fn() } };
  const render = shallow(<Auth {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    expect(render).toMatchSnapshot();
  });

  it('should call the AWS methods', () => {
    render.find('SignIn').prop('handleSignIn')();
    expect(AWSAuth.signIn).toHaveBeenCalled();
    render.find('SignUp').prop('handleSignUp')();
    expect(AWSAuth.signUp).toHaveBeenCalled();
    render.find('VerifyModal').prop('handleConfirmationCode')();
    expect(AWSAuth.confirmSignUp).toHaveBeenCalled();
  });
});
