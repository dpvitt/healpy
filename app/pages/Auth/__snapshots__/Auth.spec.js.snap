// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <SignUp
    handleSignUp={[Function]}
    setConfirmPassword={[Function]}
    setEmail={[Function]}
    setPassword={[Function]}
  />
  <SignIn
    handleSignIn={[Function]}
    setEmail={[Function]}
    setPassword={[Function]}
  />
  <VerifyModal
    handleConfirmationCode={[Function]}
    modalVisible={false}
    setConfirmationCode={[Function]}
  />
</View>
`;
